src/ledger/events.rs

```diff
@@ -0,0 +1,109 @@
+use chrono::DateTime;
+use chrono::Utc;
+use ethereum_types::U256;
+use revm::primitives::Address;
+use uuid::Uuid;
+
+use crate::eth::primitives::BlockNumber;
+use crate::eth::primitives::Hash;
+
+/// Represents token transfers (debits and credits) associated with a specific Ethereum account within a single transaction.
+///
+/// The `primary_account_address` field identifies the primary account involved in these transfers.
+///
+/// An event will be generated for each account involved in a transaction, meaning if a transaction is not the primary in this event,
+/// in another event it will treated as the primary and credit and debit operations adjusted accordingly.
+///
+/// A single event can contain multiple token transfers (e.g., a customer is debited for a card payment but receives a credit as cashback)
+pub struct AccountTransfersEvent {
+    /// ID of the event publication.
+    ///
+    /// If the event is republished, a new ID will be generated for the publication.
+    ///
+    /// Format: UUID v7 (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx)
+    pub publication_id: Uuid,
+
+    /// Datetime of the event publication.
+    ///
+    /// Format: ISO 8601
+    pub publication_datetime: DateTime<Utc>,
+
+    /// Idempotency key of the event payload (TBD: `primary_account_address` + `transaction_hash`).
+    ///
+    /// It is unique for each distinct event, but consistent across retries or republications of the same event payload.
+    ///
+    /// Format: String (TBD)
+    pub idempotency_key: String,
```

Comment:
I'd call this "event_id" and put it at the top, above "publication_id", I assume it's what clients really care about, over publication id.

Reply:
I did consider calling it just `id` or `event_id`, but some internal systems like Issuing already uses `idempotency_key` for similar scenarios. Some public APIs like Stripe and other payment providers also have fields with some variation of idempotency key.